icu4c/source/test/cintltst/cdtrgtst.c

```diff
@@ -241,6 +247,26 @@ void Test4059917()
     free(myDate);
 }
 
+UBool EqualIgnoreNumberingSystem(UChar* a, UChar* b);
+// This function return true if the sting contains the same value or the
+// numerical values are the same if we ignore the 
+UBool EqualIgnoreNumberingSystem(UChar* a, UChar* b) {
+    // len1 and len2 may not be the same if one of them are numeric outside BMP.
+    int32_t len1 = u_strlen(a);
+    int32_t len2 = u_strlen(b);
+    UChar32 ch1, ch2;
+    int32_t idx1, idx2;
+    idx1=idx2=0;
+    // U16_NEXT will increment idx1 and idx2 inside the loop.
+    while (idx1 < len1 && idx2 < len2) {
+        U16_NEXT(a, idx1, len1, ch1);
+        U16_NEXT(b, idx2, len2, ch2);
+        if (ch1 != ch2 && u_charDigitValue(ch1) != u_charDigitValue(ch2)) return false;
```

Comment:
This passes if the characters differ and neither is a digit. -->

```suggestion
        if (ch1 != ch2) {
            int32_t digit1 = u_charDigitValue(ch1);
            int32_t digit2 = u_charDigitValue(ch2);
            if (digit1 < 0 || digit1 != digit2) { return false; }
        }
```